src/components/AssetInfo/TokenSideBarContent.tsx

```diff
@@ -54,9 +68,16 @@ const TokenSideBarContent = ({
   trace(`config network: ${NETWORK_NAME} and metamask network: ${networkIdVerbose}`);
   trace(`error in sidebar access ${JSON.stringify(tokenSidebarError)}`);
   useEffect(() => {
-    setFieldValue({ ...fieldValue, ...{ approvedBeneficiary: approvedBeneficiaryAddress } });
+    setFieldValue({
+      ...fieldValue,
+      ...{
+        approvedBeneficiary: approvedBeneficiaryAddress || "",
+        approvedHolder: approvedHolderAddress || "",
+        approvedContractAddress: approvedEscrowContractAddress || ""
+      }
+    });
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
this doesn't seem to be a rule that should be disabled?

Reply:
I am using useEffect as a componentWillRecieve life cycle method. In the case it is adding fieldValue to the array which is cauing it to go into inifinite loop. find this way of doing it online though will surely search more about it.

Reply:
You mean you use a hook to behave like a deprecated method ? (https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)

What's the reason of putting the props into a local state ? looks like you can directly use the props and get rid of all the noise (sync props => state, useEffect, props/state dont have the same name)

Reply:
this is an input field state and there can be an initial value of it in some cases. so cant mutate the props..so setting the state and mutating.

Reply:
there is a problem with the flow:
- initial value comes from a very high parent component (from redux)
- this component holds a state based on the parent (never a good idea)
- the state is then updated by child component using an indirect link => `setFieldValue({ ...fieldValue, ...{ [e.target.name]: e.target.value } });`, the name of the child component form must match the name of the parent component field state.

I would suggest to create separate state instead of holding everything into an object, with separate functions to update (why passing each value down individually and not as an object, but only one function ?).
I would also suggest to create a context somewhere to avoid those intermediate state and have one single source of truth for this, because for the moment it's widely split.

Additional question => In which scenario the props is likely to change ?

Reply:
it's an async call to fetch the holder and beneficiary value. So the initial state value is an empty string. and when the value is fetched from the blockchain then the props get updated.

Reply:
Also would it be better to use useSelector to access the state in component instead of context ?

Reply:
Maybe shouldnt display the component while it's loading ?

I dont mind context or selector as long as the state is not split into multiple components with indirect coupling.

Reply:
yea need to put loader while fetching data. have to extract out loader code in a separate component also. will work on another PR. already made an issue for that. made some changes for state refactoring in latest commit. Please check.